### PR TITLE
Fix bugs in the Sample Transmission Calculator

### LINKED ISSUE #35011: Minor Sample Transmission Calculator Improvements
**Describe the bug**
 - [ ] A) Missing axes labels on the embedded plot
 - [ ] B) Single wavelength: Width is able to be larger than the max
 - [ ] C) Multiple wavelength: Only allows odd number of wavelengths without explanation

**To Reproduce**
A)
1. Open the GUI and check the Axes

B)
1. Set the min to 1, max to 10, and the width to 12. 
2. Enter `H2 O` into the Chemical Formula field
4. Click calculate
5. No points will be present

C)
1. Select Multiple from the dropdown combobox
2. Enter 1,2,3,4 
3. Click calculate
4. An error will appear

**Expected behavior**
A) Labels present
B) A warning/error message is shown
C) Explain in the error message why an odd number is required (Needs to be in point, width, point, width, point, etc.)

**Platform/Version (please complete the following information):**
 - OS: All
 - Mantid Version : 6.6

### LINKED ISSUE #35586: Usability improvements for Sample Transmission Calculator
The following potential usability improvements were identified during manual testing for release 6.7:

- [ ] It was felt that the visibility of the error messages could potentially be improved, as it was easy to miss the red message at the bottom of the dialog. One suggestion was to provide pop-up error messages, although we should consider whether some users may find these irritating.
- [ ] A tooltip to explain the format to enter into the Multiple input wavelength range type (which takes a rebin parameter string) would be helpful, as we didn't find this very intuitive.
- [ ] Including axis labels on the plot might be helpful.

See https://docs.mantidproject.org/interfaces/general/Sample%20Transmission%20Calculator.html for information on the tool.

### PR DESCRIPTION
**Pull request by jclarkeSTFC**
**Description of work**
A few bug fixes and usability improvements for the Sample Transmission Calculator.

**Purpose of work**

Fixes #35011 and fixes #35586

**Summary of work**

- Added tool tip to edit box when using the Multiple option
- Added labels to the axes
- Add error if the bin width is larger than the max
- Highlighted error messages a bit more with a red box

**To test:**
See #35011 for how to reproduce.

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/python/mantidqtinterfaces/test/sample_transmission_calculator/stc_model_test.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def test_validate_histogram_single_max_lower_than_min(self):
         input_dict = {
             "binning_type": 0,
             "single_low": 1.0,
-            "single_width": 0.1,
+            "single_width": 0.0,
```
Comment (jhaigh0) [inline qt/python/mantidqtinterfaces/test/sample_transmission_calculator/stc_model_test.py:101]: What was the reason for changing this value?

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/SampleTransmissionCalculator/stc_view.py [hunk 2]
```diff
@@ -19,13 +19,21 @@ def __init__(self, parent=None):
         fig = Figure()
         self.axes = fig.add_subplot(111)
         self.plot_frame = FigureCanvas(fig)
```
Comment (jhaigh0) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/SampleTransmissionCalculator/stc_view.py:21]: When resizing to a small size, the axis labels can run off the edge of the plot area and become hidden. Giving the figure a tight layout is the simplest way to fix this.

```suggestion
        fig = Figure()
        fig.set_layout_engine(layout="tight")
        self.axes = fig.add_subplot(111)
        self.plot_frame = FigureCanvas(fig)
```

**Review by jhaigh0 — CHANGES_REQUESTED** [review]
Nice improvements to the feedback of the interface :)

One further thing I wondered about was changing where the 'expanding areas' of the gui are for single input.
image

At the moment when you expand the window it adds space inbetween say `Width:` and the width box. I think these expansions should happen inbetween the previous box and the next lable. So labels stay attached to there respective boxes.

File: qt/python/mantidqtinterfaces/test/sample_transmission_calculator/stc_model_test.py (hunk 1 above)
Comment (jclarkeSTFC) [reply qt/python/mantidqtinterfaces/test/sample_transmission_calculator/stc_model_test.py:101]: The test name suggests that it's checking max < min, but it also had width > max, which would have generated two errors, so I added a separate test for checking the width > max case. The options for this case to make the test pass were to either change the width or change the expected error message, so I changed the width.

**Review by jhaigh0 — APPROVED** [review]
Thanks for making the changes 👍

**Pull request merged**